src/metabase/search/filter.clj

```diff
@@ -132,7 +132,7 @@
 (defn- default-created-at-filter-clause
   [model created-at]
   (let [{:keys [start end]} (try
-                             (params.dates/date-string->range created-at {:inclusive-end? false})
+                             (params.dates/date-string->range created-at {:inclusive-end? false :timezone-id "UTC"})
                              (catch Exception _e
                                (throw (ex-info (tru "Failed to parse created at param: {0}" created-at) {:status-code 400}))))
         start               (some-> start u.date/parse)
```

Comment:
And I guess you're going to convert these from the timezone-id time zone to the system time zone again, before the filter is applied?

Reply:
so you're right we should make sure the param for the filter shouldn't be in LocalDateTime.

And the reason for that is when we execute the statement, `jdbc/prepare` will automatically convert `LocalDateTime` to a `ZonedDateTime`, assuming the timezone is system timezone.

To avoid this I've changed so that the datetime param is a ZonedDateTime.


```
printing the generated stmt from toucan2.jdbc.query/reduce-jdbc-query

In this setup, I've changed my system timezone to Aisa/Tokyo (UTC +9)

(t2/select-one-pk :model/Card :created_at #t "2023-08-11T00:00:00")
#<com.mchange.v2.c3p0.impl.NewProxyPreparedStatement@6cb6eba3 com.mchange.v2.c3p0.impl.NewProxyPreparedStatement@6cb6eba3 [wrapping: SELECT * FROM "report_card" WHERE "created_at" = '2023-08-11 00:00:00+09'::timestamp]>



(t2/select-one-pk :model/Card :created_at #t "2023-08-11T00:00:00Z")
#<com.mchange.v2.c3p0.impl.NewProxyPreparedStatement@7b27250c com.mchange.v2.c3p0.impl.NewProxyPreparedStatement@7b27250c [wrapping: SELECT * FROM "report_card" WHERE "created_at" = '2023-08-11 00:00:00+00'::timestamp with time zone]>

```